zkevm-circuits/src/bytecode_circuit/dev.rs

```diff
@@ -30,29 +24,35 @@ impl<F: Field> Circuit<F> for BytecodeCircuitTester<F> {
 
     fn configure(meta: &mut ConstraintSystem<F>) -> Self::Config {
         let bytecode_table = BytecodeTable::construct(meta);
-
-        let randomness = power_of_randomness_from_instance::<_, 1>(meta);
         let keccak_table = KeccakTable::construct(meta);
+        let challenges = Challenges::construct(meta);
 
-        Config::configure(meta, randomness[0].clone(), bytecode_table, keccak_table)
+        let config = {
+            let challenges = challenges.exprs(meta);
+            Config::configure(meta, bytecode_table, keccak_table, challenges)
+        };
+
+        (config, challenges)
     }
 
     fn synthesize(
         &self,
-        config: Self::Config,
+        (config, challenges): Self::Config,
         mut layouter: impl Layouter<F>,
     ) -> Result<(), Error> {
+        let challenges = challenges.values(&mut layouter);
```

Comment:
If I'm not mistaken, the challenges used in the mock prover are deterministic and random looking (generated here https://github.com/han0110/halo2/blob/346afce254258361f65a4e298f8636a52871cb6d/halo2_proofs/src/dev.rs#L544)

Recently while debugging a lookup error it was useful to me to use randomness `0x100` so that I could see the original values after an RLC (where the original values were smaller than 256 each).  I think it would be useful to allow specifying custom challenge values for the mock proving for debugging.

Reply:
This makes sense, but this requires to update `pse/halo2`, I have created an feature request issue https://github.com/privacy-scaling-explorations/halo2/issues/101 for this.